internal/webhook/manager.go

```diff
@@ -0,0 +1,479 @@
+						},
+					},
+				},
+				nil,
+			),
+		},
+	}
+
+	vwc.Webhooks = append(vwc.Webhooks, getDesiredKueueValidatingWebhooks(namespace, KueueManagedLabelKey, "")...)
+	// TODO: Remove this once we drop support for the legacy label
+	vwc.Webhooks = append(vwc.Webhooks, getDesiredKueueValidatingWebhooks(namespace, KueueLegacyManagedLabelKey, "-legacy")...)
+
+	return vwc
+}
+
+// createOrUpdateWebhookConfiguration handles the creation or update of Mutating/ValidatingWebhookConfiguration.
+//
+// Parameters:
+//   - ctx: The context for the API call
+//   - c: The client to use for the API call
+//   - log: The logger to use for the API call
+//   - desiredWebhookConfig: The desired webhook configuration to create or update
+//
+// Returns:
+//   - error: Any error encountered while creating or updating the webhook configuration
+func createOrUpdateWebhookConfiguration(ctx context.Context, c client.Client,
+	log logr.Logger, desiredWebhookConfig client.Object) error {
+	foundWebhookConfig, ok := desiredWebhookConfig.DeepCopyObject().(client.Object)
+	if !ok {
+		return errors.New("failed to deep copy object")
+	}
+
+	err := c.Get(ctx, client.ObjectKeyFromObject(desiredWebhookConfig), foundWebhookConfig)
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			log.Info(fmt.Sprintf("Creating %s", desiredWebhookConfig.GetName()), "Name", desiredWebhookConfig.GetName())
+			if err := c.Create(ctx, desiredWebhookConfig); err != nil {
+				log.Error(err, fmt.Sprintf("Failed to create %s", desiredWebhookConfig.GetName()), "Name", desiredWebhookConfig.GetName())
+				return err
+			}
+		} else {
+			log.Error(err, fmt.Sprintf("Failed to get %s", desiredWebhookConfig.GetName()), "Name", desiredWebhookConfig.GetName())
+			return err
+		}
+	} else {
+		// Compare found and desired webhook configurations' webhooks list for updates
+		needsUpdate := false
+		switch desired := desiredWebhookConfig.(type) {
+		case *admissionregistrationv1.MutatingWebhookConfiguration:
+			found, ok := foundWebhookConfig.(*admissionregistrationv1.MutatingWebhookConfiguration)
+			if !ok {
+				return errors.New("failed to cast webhook configuration object to MutatingWebhookConfiguration")
+			}
+			if !reflect.DeepEqual(found.Webhooks, desired.Webhooks) {
+				needsUpdate = true
+			}
+		case *admissionregistrationv1.ValidatingWebhookConfiguration:
+			found, ok := foundWebhookConfig.(*admissionregistrationv1.ValidatingWebhookConfiguration)
+			if !ok {
+				return errors.New("failed to cast webhook configuration object to ValidatingWebhookConfiguration")
+			}
+			if !reflect.DeepEqual(found.Webhooks, desired.Webhooks) {
+				needsUpdate = true
+			}
+		default:
+			log.Error(errors.New("unsupported webhook configuration type"), "Cannot compare webhook config", "Kind", desiredWebhookConfig.GetObjectKind().GroupVersionKind().Kind)
+			return fmt.Errorf("unsupported webhook configuration type %T", desiredWebhookConfig)
+		}
+
+		if needsUpdate {
```

Comment:
Not sure I would use `reflect.DeepEqual`  to figure out if a webhook needs update since the webhook is amended by the ca bundle injector, which is one of the reason I suggest to evaluate SSA and just update it on each reconcile.